numba/tests/test_tuples.py

```diff
@@ -667,20 +667,16 @@ def check(p):
         with self.assertRaises(errors.TypingError) as raises:
             check([4, 5])
 
-        if utils.PYVERSION > (3, 8):
-            # Python 3.9 has a peephole rewrite due to large changes in tuple
-            # unpacking. It results in a tuple + list situation from the above
-            # so the error message reflects that. Catching this specific and
-            # seemingly rare sequence in the peephole rewrite is prohibitively
-            # hard. Should it be reported numerous times, revisit then.
-            msg1 = "No implementation of function"
-            self.assertIn(msg1, str(raises.exception))
-            msg2 = "tuple(reflected list(" # ignore the rest of reflected list
-                                           # part, it's repr is quite volatile.
-            self.assertIn(msg2, str(raises.exception))
-        else:
-            msg = "Only tuples are supported when unpacking a single item"
-            self.assertIn(msg, str(raises.exception))
+        # Python 3.9 has a peephole rewrite due to large changes in tuple
```

Comment:
Is this 3.9 only or does it affect other Python versions too? If it is 3.9 only, perhaps it should go into a `if PYVERSION` block? If this is more Pythons then perhaps:

```suggestion
        # Python 3.9+ has a peephole rewrite due to large changes in tuple
```

Reply:
As agreed with @stuartarchibald this will go into it's own PR.